### PR TITLE
Fix re-transmission table size is 0 if Lwip custom pool is used

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* We found   re-transmission table size is 0 on one of products when we integrate Matter stack.
 
```
#ifdef PBUF_POOL_SIZE
#define CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE std::min(PBUF_POOL_SIZE, CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS)
#elif CHIP_SYSTEM_CONFIG_PACKETBUFFER_POOL_SIZE != 0
#define CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE std::min(CHIP_SYSTEM_CONFIG_PACKETBUFFER_POOL_SIZE, CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS)
```

This block of code will result in a 0 sized retransmit table if you have PBUF_POOL_SIZE set to 0. Which is what you are instructed to do in the configuration header for LWIP if you set LWIP_PBUF_FROM_CUSTOM_POOLS to 1.

This resulted in us not being able to use RMP because our product is using custom pbuf pools. We crashed pretty quickly.

#### Change overview
Fix re-transmission table size is 0 if Lwip custom pool is used

#### Testing
How was this tested? (at least one bullet point required)
* confirm CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE is not 0 if LWIP_PBUF_FROM_CUSTOM_POOLS is set to 1

### CODE REVIEW

**Review by tcarmelveilleux — APPROVED** [review]

File: src/messaging/ReliableMessageProtocolConfig.h [hunk 1]
```diff
@@ -90,7 +90,7 @@ namespace Messaging {
  *
  */
 #ifndef CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE
-#ifdef PBUF_POOL_SIZE
+#if !LWIP_PBUF_FROM_CUSTOM_POOLS && PBUF_POOL_SIZE
```
Comment (tcarmelveilleux) [inline src/messaging/ReliableMessageProtocolConfig.h:93]: I would in general argue for this config to ALWAYS be set **per product**, and not try to make assumptions based on any particular usage of lwIP...

**Review by saurabhst — APPROVED** [review]

File: src/messaging/ReliableMessageProtocolConfig.h (hunk 1 above)
Comment (kghost) [reply src/messaging/ReliableMessageProtocolConfig.h:93]: It looks like there is a quite complex logic to choose the `CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE`. I'm not a fun of it.

**Review by kghost — COMMENTED** [review]
Can we add a `static_assert` in `src/messaging/ReliableMessageMgr.h` to ensure that this will never happen again ?

File: src/messaging/ReliableMessageProtocolConfig.h (hunk 1 above)
Comment (yufengwangca) [reply src/messaging/ReliableMessageProtocolConfig.h]: Yes, this logic is indeed looks like a puzzle, try to simply it as much as I can. 

I agree with @tcarmelveilleux there is no guarantee config LWIP_PBUF_FROM_CUSTOM_POOLS is always set for each product. so we just define CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE to CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS only if this config flag is defined and not set to 0.

Comment (yufengwangca) [reply src/messaging/ReliableMessageProtocolConfig.h]: Changed #ifdef PBUF_POOL_SIZE to #elif PBUF_POOL_SIZE which should have similar effect as static_assert check

**Review by woody-apple — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**